stellargraph/data/unsupervisedSampler.py

```diff
@@ -0,0 +1,229 @@
+               number_of_walks <int>: Number of walks from each root node. Note, at the moment this parameter is only used to calculate the number of batches in each epoch,
+                                      as we do not have a pregenerated sample list, we need to estimate given a batch_size, how many batches to generate in each epoch.  
+               
+        
+        """
+        # Initialize the random state
+        self._random_state = random.Random(seed)
+
+        # Initialize a numpy random state (for numpy random methods)
+        self._np_random_state = np.random.RandomState(seed=seed)
+
+        if not isinstance(G, StellarGraphBase):
+            raise ValueError(
+                "({}) Graph must be a StellarGraph object.".format(type(self).__name__)
+            )
+        else:
+            self.graph = G
+
+        # walker
+        if walker is not None:
+            if not isinstance(
+                walker, UniformRandomWalk
+            ):  # only work with Uniform Random Walker at the moment
+                raise TypeError(
+                    "({}) Only Uniform Random Walks are possible".format(
+                        type(self).__name__
+                    )
+                )
+            else:
+                self.walker = UniformRandomWalk(G)
+        else:
+            self.walker = UniformRandomWalk(G)
+
+        if nodes is None:
+            self.nodes = (
+                G.nodes()
+            )  # if no root nodes are provided for sampling defaulting to using all nodes as root nodes.
+        elif not is_real_iterable(nodes):  # check whether the nodes provided are valid.
+            raise ValueError("nodes parameter should be an iterableof node IDs.")
+        else:
+            self.nodes = nodes
+
+        if (
+            length < 2
+        ):  # for a  sample pair need at least one neighbor beyond the root node.
+            raise ValueError(
+                "({}) For generating (target,context) samples, walk length has to be at least 2".format(
+                    type(self).__name__
+                )
+            )
+        else:
+            self.length = length
+
+        if number_of_walks < 1:
+            raise ValueError(
+                "({}) At least 1 walk from each head node has to be done".format(
+                    type(self).__name__
+                )
+            )
+        else:
+            self.number_of_walks = number_of_walks
+
+        self.seed = seed
+
+    def generator(self, batch_size):
+
+        """
+        This method yields a batch_size of +/- samples when it is called. The batch_size must be provided. It should be a positive integer. Moreover, for now we make the batch_size even by
+        unit incrementing it, incase it is odd. This is because we generate one negative sample for each positive sample.
+        This method:
+            1. Iterates over the root nodes.
```

Comment:
Don't use numbered lists in API docstrings as Sphinx doesn't work well with them.

Reply:
Fixed the comments.